recipes/msys2/all/conanfile.py

```diff
@@ -208,6 +208,8 @@ def _do_build(self):
         with tools.chdir(os.path.join(self._msys_dir, "usr", "bin")):
             for package in packages:
                 self.run('bash -l -c "pacman -S %s --noconfirm"' % package)
+            for package in ['pkgconf']:
+                self.run('bash -l -c "pacman -Rs $(pacman -Qsq %s) --noconfirm"' % package)
```

Comment:
```suggestion
            for package in ['pkgconf', 'autoconf', 'automake', 'gettext-devel', 'libiconv-devel', 'libtool', 'make']:
                self.run('bash -l -c "pacman -Rs $(pacman -Qsq ^%s) --noconfirm"' % package)
```

these are other packages provided by conan. Do you have objections to removing these too ?

Reply:
No objection, but maybe list & fix all CCI recipes silently relying on autotools & pkgconf before merging this PR.

Reply:
What's the content of `/usr/lib(64)/pkgconfig`?
Perhaps this folder can be removed too?

Reply:
```
$ ls /usr/lib/pkgconfig/
autoopts.pc  libmagic.pc          python-3.8.pc     python3.pc
libalpm.pc   python-3.8-embed.pc  python3-embed.pc  tcl.pc
```
```
$ ls /usr/share/pkgconfig/
bash-completion.pc  libmakepkg.pc
```

Reply:
Let's keep the folder and only remove it (or its contents) only when it is starting to cause problems.

Reply:
Removing `make` from msys2 could break MiNGW builds of many recipes I think.

Reply:
To sum this up, I think this is too big of a change for now. (removing pkgconf should be safe, because It was not [present in precedent versions of msys](https://github.com/msys2/MSYS2-packages/commit/3009e7e836cc46b0ff4a9832f49741c66d1cfff7#diff-ca27ad8b0bbc2ac3072fcc97ec32b241e461331d0baef59595a03cff6ad6f013))